kokkos ecosystem: v4.7.01

ndellingwood opened the pull request:
Updating the versions for the 4.7.01 patch release.

<!--  
Remember that `spackbot` can help with your PR in multiple ways:
- `@spackbot help` shows all the commands that are currently available
- `@spackbot fix style` tries to push a commit to fix style issues in this PR
- `@spackbot re-run pipeline` runs the pipelines again, if you have write access to the repository 
-->

ndellingwood committed:
Kokkos Ecosystem: release of version 4.7.01

Updating the versions for the 4.7.01 patch release.

Signed-off-by: Nathan Ellingwood <ndellin@sandia.gov>

ndellingwood commented:
I'm unable to add/request reviewers, pinging a few people active with the kokkos ecosystem package support:
@cedricchevalier19 @lucbv @crtrott @masterleinad @rbberger

cedricchevalier19 commented on repos/spack_repo/builtin/packages/kokkos/package.py:
```diff
@@ -445,7 +446,7 @@ def cmake_args(self):
         )
 
-        if self.version == Version("4.7.00"):
+        if self.version == Version("4.7.00") or self.version == Version("4.7.01"):
```
This conditional was introduced as a temporary fix until 4.7.01, is it necessary to extend it to this new patch release?

tpadioleau replied:
It does not seem necessary as the release 4.7.01 now defaults `Kokkos_ENABLE_IMPL_VIEW_LEGACY` to `True`, see https://github.com/kokkos/kokkos/blob/4.7.01/CHANGELOG.md#4701

ndellingwood replied:
thanks @cedricchevalier19 @tpadioleau , I dropped the 4.7.01 version check in [f9f0c71](https://github.com/spack/spack-packages/pull/1727/commits/f9f0c71ce3d34adf47b210cb0ce89b33fcd871a5)

tpadioleau reviewed (changes requested):
Thanks for the update, as the Kokkos release 4.7.01 also fixes build issues with CUDA 13, I would suggest to add a conflict with CUDA >=13 for Kokkos <4.7.1, in terms of Spack it should be `conflicts("^cuda@13:", when="@:4.7.0")`.

ndellingwood committed:
address reviewer feedback

- drop self.version check for 4.7.01
- add conflict for cuda/13 with version < 4.7.01

Signed-off-by: Nathan Ellingwood <ndellin@sandia.gov>

ndellingwood commented:
> I would suggest to add a conflict with CUDA >=13 for Kokkos <4.7.1, in terms of Spack it should be `conflicts("^cuda@13:", when="@:4.7.0")`.

@tpadioleau thanks for the suggestion and spack snippet, I added this in [f9f0c71](https://github.com/spack/spack-packages/pull/1727/commits/f9f0c71ce3d34adf47b210cb0ce89b33fcd871a5) , let me know if there is a better location in the package file where this should be placed

lucbv reviewed (approved):
Thanks @ndellingwood will keep an eye on the last CI check and ping someone to merge once it passes

tpadioleau reviewed (approved):
Looks ok to me, thank you.

By the way, this makes me realize that the conflict on cmake 3.28 is likely not working as expected. I can open a PR later.

cedricchevalier19 reviewed (approved):
Thank you!

lucbv commented:
@rbberger could you have a look and maybe merge this? Thx : )

tpadioleau commented:
> Looks ok to me, thank you.
> 
> By the way, this makes me realize that the conflict on cmake 3.28 is likely not working as expected. I can open a PR later.

see https://github.com/spack/spack-packages/pull/1766
